lib/node_modules/@stdlib/ndarray/base/every-by/lib/main.js

```diff
@@ -0,0 +1,298 @@
+var every0d = require( './0d.js' );
+var every1d = require( './1d.js' );
+var every2d = require( './2d.js' );
+var every3d = require( './3d.js' );
+var everynd = require( './nd.js' );
+
+
+// VARIABLES //
+
+var EVERY = [
+	every0d,
+	every1d,
+	every2d,
+	every3d
+];
+var ACCESSOR_EVERY = [
+	accessorevery0d,
+	accessorevery1d,
+	accessorevery2d,
+	accessorevery3d
+];
+var COMPLEX_EVERY = [
+	complexevery0d,
+	complexevery1d,
+	complexevery2d,
+	complexevery3d
+];
+var BLOCKED_EVERY = [
+	blockedevery2d, // 0
+	blockedevery3d
+];
+var BLOCKED_ACCESSOR_EVERY = [
+	blockedaccessorevery2d, // 0
+	blockedaccessorevery3d
+];
+var BLOCKED_COMPLEX_EVERY = [
+	blockedcomplexevery2d, // 0
+	blockedcomplexevery3d
+];
+var MAX_DIMS = EVERY.length - 1;
+
+
+// MAIN //
+
+/**
+* Tests whether every element in an ndarray is truthy according to a callback function.
+*
+* ## Notes
+*
+* -   A provided ndarray should be an `object` with the following properties:
+*
+*     -   **dtype**: data type.
+*     -   **data**: data buffer.
+*     -   **shape**: dimensions.
+*     -   **strides**: stride lengths.
+*     -   **offset**: index offset.
+*     -   **order**: specifies whether an ndarray is row-major (C-style) or column major (Fortran-style).
+*
+* @param {ArrayLikeObject<Object>} arrays - array-like object containing one input array
+* @param {Function} clbk - callback function
+* @param {thisArg} thisArg - callback execution context
+* @returns {boolean} result
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* function clbk( value ) {
+*    return value > 0.0;
+* }
+*
+* // Create a data buffer:
+* var xbuf = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0 ] );
+*
+* // Define the shape of the input array:
+* var shape = [ 3, 1, 2 ];
+*
+* // Define the array strides:
+* var sx = [ 4, 4, 1 ];
+*
+* // Define the index offset:
+* var ox = 1;
+*
+* // Create the input ndarray-like object:
+* var x = {
+*     'ref': null,
+*     'dtype': 'float64',
+*     'data': xbuf,
+*     'shape': shape,
+*     'strides': sx,
+*     'offset': ox,
+*     'order': 'row-major'
+* };
+*
+* // Test elements:
+* var out = everyBy( [ x ], clbk );
+* // returns true
+*/
+function everyBy( arrays, clbk, thisArg ) {
+	var isCmplx;
+	var ndims;
+	var xmmv;
+	var shx;
+	var iox;
+	var len;
+	var sx;
+	var ox;
+	var ns;
+	var x;
+	var d;
+	var i;
+
+	// Unpack the ndarray and standardize ndarray meta data:
+	x = ndarray2object( arrays[ 0 ] );
+	shx = x.shape;
+	ndims = shx.length;
+
+	// Check for known array types which can be reinterpreted for better iteration performance...
+	if ( isBooleanArray( x.data ) ) {
+		x.data = reinterpretBoolean( x.data, 0 );
+		x.accessorProtocol = false;
+	} else if ( isComplexArray( x.data ) ) {
+		x.data = reinterpretComplex( x.data, 0 );
+		x.accessorProtocol = false;
+		x.strides = gscal( ndims, 2, x.strides, 1 );
+		x.offset *= 2;
+		isCmplx = true;
+	}
```

Comment:
This should be removed.

Reply:
When a user provides a boolean array, they expect the callback to be provided a boolean value, not a number. Similarly for complex number arrays. As such, no choice but to use the normal accessor kernels.